guides/v2.1/ui_comp_guide/concepts/ui_comp_data_source.md

```diff
@@ -0,0 +1,113 @@
+---
+layout: default
+group: UI_Components_guide
+subgroup: concepts
+title: The DataSource Component
+menu_title: Providing Data to UI Components
```

Comment:
For the title: and subgroup: values....  
... I'm unsure if this new topic fits best under the Concepts section, and thus should be called "About the DataSource Component" or if it should be under the planned How To section... Which doesn't yet exist but is easily added by using a value of subgroup: How To.

Let me know your thoughts on that, and let me consult with @almarchenko who "owns" this book, and we can decide before we merge the PR. ;-) I tend to lean towards having it as a conceptual topic, even though it has a lot of how to steps in it.